### PR TITLE
make sure the shape matches the size of array in `to_array`

### PR DESCRIPTION
**Pull request by kitchoi**
This is for issue #197 
@prabhuramachandran, @dmsurti  can you take a look please?

### CODE REVIEW

**Comment by dmsurti** [comment]
@kitchoi Can you add a test in `tvtk/tests/test_array_handler.py`?

**Comment by kitchoi** [comment]
@dmsurti Sure.

**Comment by kitchoi** [comment]
@dmsurti I am wondering if the cache should be cleared in the same place where I added the shape check.  The cache is there to prevent the user from modifying the array being sent to the vtk, but this PR is allowing the opposite.  In other words, is the cache still needed?  (i can't find `array_cache.get` anywhere else)

**Comment by dmsurti** [comment]
> The cache is there to prevent the user from modifying the array being sent to the vtk, but this PR is allowing the opposite

Nope. The cache is to prevent only from deleting/resizing. https://github.com/enthought/mayavi/blob/master/tvtk/array_handler.py#L61.

File: tvtk/tests/test_array_handler.py [hunk 1]
```diff
@@ -396,6 +396,20 @@ def test_array_cache(self):
...
+        arr  = numpy.arange(8).reshape(2, 4)
+        vtk_arr = array_handler.array2vtk(arr)
+        arr1 = array_handler.vtk2array(vtk_arr)
+
+        # the vtk array is appended, shapes don't match cached array anymore
+        vtk_arr.InsertTuple4(2, 1., 2., 3., 4.)
+
+        # arr2 has a different shape
+        arr2 = array_handler.vtk2array(vtk_arr)
+        self.assertEqual(arr2.shape, (3, 4))
```
Comment (itziakos) [inline tvtk/tests/test_array_handler.py]: you should also check the values of the array to make sure that all is as expected

Comment (kitchoi) [inline tvtk/tests/test_array_handler.py]: Good point.  Should inserting float values to a vtk array originally initialised with int values trigger a type casting?
(Separate question, what is the lowest version requirement of numpy for mayavi?)

**Comment by prabhuramachandran** [comment]
@kitchoi -- thanks for catching this!  LGTM.

**Pull request merged**